internal/cluster/watch.go

```diff
@@ -21,46 +23,46 @@ var (
 )
 
 // WatchPipelineActivity watches the jx namespace for changes to PipelineActivities.
-func WatchPipelineActivity(done chan struct{}, namespace string, jxClient *jenkinsclientv1.JenkinsV1Client) {
-	listWatch := cache.NewListWatchFromClient(jxClient.RESTClient(), "pipelineactivities", namespace, fields.Everything())
+func WatchPipelineActivity(done chan struct{}, namespace string, restClient rest.Interface, pipelineActivitiesGetter jenkinsclientv1.PipelineActivitiesGetter) {
+	listWatch := cache.NewListWatchFromClient(restClient, "pipelineactivities", namespace, fields.Everything())
 	kube.SortListWatchByName(listWatch)
 	_, actController := cache.NewInformer(
 		listWatch,
 		&jenkinsv1.PipelineActivity{},
 		time.Minute*10,
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
-				onPipelineActivity(obj, jxClient)
+				onPipelineActivity(obj, pipelineActivitiesGetter)
 			},
 			UpdateFunc: func(oldObj, newObj interface{}) {
-				onPipelineActivity(newObj, jxClient)
+				onPipelineActivity(newObj, pipelineActivitiesGetter)
 			},
 			DeleteFunc: func(obj interface{}) {},
 		},
 	)
-	go actController.Run(done)
+	actController.Run(done)
 }
 
-func onPipelineActivity(obj interface{}, jxClient *jenkinsclientv1.JenkinsV1Client) {
+func onPipelineActivity(obj interface{}, pipelineActivitiesGetter jenkinsclientv1.PipelineActivitiesGetter) {
 	pipelineActivity, ok := obj.(*jenkinsv1.PipelineActivity)
 	if !ok {
 		logger.Warnf("unexpected type %T", obj)
 	} else {
-		err := handlePipelineActivity(pipelineActivity, jxClient)
+		err := handlePipelineActivity(pipelineActivity, pipelineActivitiesGetter)
 		if err != nil {
 			logger.Error(err)
 		}
 	}
 }
 
-func handlePipelineActivity(pipelineActivity *jenkinsv1.PipelineActivity, jxClient *jenkinsclientv1.JenkinsV1Client) (err error) {
+func handlePipelineActivity(pipelineActivity *jenkinsv1.PipelineActivity, pipelineActivitiesGetter jenkinsclientv1.PipelineActivitiesGetter) (err error) {
 	for _, attachment := range pipelineActivity.Spec.Attachments {
 		if attachment.Name != "jacoco" {
 			continue
 		}
 
-		// TODO Handle having multiple attachments properly
 		for _, url := range attachment.URLs {
+			//  append version string to report URL to avoid any caching issues when retrieving the report
```

Comment:
Should this not rather be done via `Cache-Control` headers in the `ReadURL` function, or even in a specific bucket provider known to suffer from caching issues?